src/algorithms/reco/JetReconstruction.cc

```diff
@@ -23,31 +23,32 @@ namespace eicrecon {
     m_log->trace("Initialized");
   }
 
-  std::vector<edm4eic::ReconstructedParticle*> JetReconstruction::execute(
+  edm4eic::ReconstructedParticleCollection* JetReconstruction::execute(
     const std::vector<const edm4hep::LorentzVectorE*> momenta) {
 
     // Store the jets
-    std::vector<edm4eic::ReconstructedParticle*> jets_edm;
+    edm4eic::ReconstructedParticleCollection* jet_collection = new edm4eic::ReconstructedParticleCollection();
 
     // Skip empty
     if (momenta.empty()) {
       m_log->trace("  Empty particle list.");
-      return jets_edm;
+      return jet_collection;
     }
 
     m_log->trace("  Number of particles: {}", momenta.size());
 
     // Particles for jet reconstrution
     std::vector<PseudoJet> particles;
     for (const auto &mom : momenta) {
+
       // Only cluster particles within the given pt Range
       if ((mom->pt() > m_minCstPt) && (mom->pt() < m_maxCstPt)) {
-        particles.emplace_back(mom->px(), mom->py(), mom->pz(), mom->e() );
+        particles.push_back( PseudoJet(mom->px(), mom->py(), mom->pz(), mom->e()) );
```

Comment:
You are reverting changes in the target branch.

Reply:
Ah! Thank you for catching that!